compiler/optimizer/OMROptimizationManager.cpp

```diff
@@ -49,7 +49,7 @@ OMR::OptimizationManager::OptimizationManager(TR::Optimizer *o, OptimizationFact
         _numPassesCompleted(0),
         _optData(NULL),
         _optPolicy(NULL),
-        _enabled(false),
+        _enabled(!o->comp()->isDisabled(optNum)),
```

Comment:
My main concern in regards to compile-time was just recent experience with self() calls in extensible classes where the self() wasn't inlined leading to a hard-to-find compile-time increase. You may be right in this case since in many cases it is going to replace existing calls and could memoize some of the costs of the existing checking logic. To be honest most of the optimizer changes do a have a compile-time verification these days to avoid creep over time, but I agree it probably isn't strictly necessary as long as the number of indirects per query etc looks to be consistent for the most part.